Cargo.toml

```diff
@@ -14,12 +14,12 @@ homepage = "http://github.com/cgwalters/openat-ext"
 documentation = "http://docs.rs/openat-ext"
 
 [dependencies]
-drop_bomb = "0.1.5"
-openat = "0.1.15"
-libc = "0.2.34"
-nix = "0.18"
-rand = "0.7.3"
+drop_bomb = ">= 0.1.5, < 0.2"
```

Comment:
I'm a bit confused because I thought these were equivalent in Cargo?  See the docs: https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-cratesio

Basically `drop_bomb = "0.1.5"` is shorthand for `drop_bomb = "^0.1.5"` which in turn expands to exactly this range.

AFAICS you are keeping the same versions here.  Is there a specific crate in Fedora this is failing on?

Reply:
The two declarations are exactly equivalent in Cargo.  The difference is that Dependabot won't touch these unless a new release falls outside the specified range.  Otherwise, Dependabot would e.g. bump libc 0.2.34 to 0.2.88, and then we'd have to patch the dep downstream if Fedora was still shipping 0.2.85.

Dependabot knows how to widen dependency ranges if it finds them, but for Rust it currently [can't be configured](https://dependabot.com/docs/config-file/#version_requirement_updates) to do that unconditionally.  The workaround is to specify ranges explicitly.

**However**: as it turns out, the above only applies to applications.  Dependabot _does_ do the right thing for libraries.  I'll close.

Reply:
Ahhh I see.